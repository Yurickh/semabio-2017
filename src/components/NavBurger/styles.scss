.nav-burger {
  position: relative;

  width: 3rem;
  height: 3rem;

  cursor: pointer;

  &::before,
  &::after {
    position: absolute;

    display: block;

    width: 20px;
    height: 2px;

    content: '';
    transition: all .3s ease;

    background-color: currentColor;
  }

  &::before {
    top: 17px;
    left: 13px;

    box-shadow: 0 12px 0 0 currentColor;
  }

  &::after {
    top: 23px;
    left: 13px;
  }

  &.-open {
    position: fixed;
    top: 30px;
    right: 22%;
    z-index: 2;

    &::before,
    &::after {
      top: 23px;
      left: 11px;

      width: 24px;
    }

    &::before {
      transform: rotate(45deg);

      box-shadow: 0 0 0 0 transparent;
    }

    &::after {
      transform: rotate(-45deg);
    }
  }
}
