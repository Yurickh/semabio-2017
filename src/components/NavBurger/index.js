import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './styles.css'

class NavBurger extends Component {
  render() {
    return (
      <div className={ `nav-burger ${this.props.variant}` }
        onClick={this.props.onClick}
      />
    )
  }
}

NavBurger.propTypes = {
  variant: PropTypes.oneOf(['-open', '']),
  onClick: PropTypes.func.isRequired,
}

NavBurger.defaultProps = {
  variant: '',
}

export default NavBurger
