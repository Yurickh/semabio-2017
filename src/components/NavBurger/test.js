import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import NavBurger from '../NavBurger'

describe('NavBurger', () => {
  let burger
  let onClick

  beforeEach(() => {
    onClick = jest.fn()

    burger = shallow(
      <NavBurger onClick={onClick} />
    )
  })

  it('should render correctly', () => {
    expect(toJSON(burger)).toMatchSnapshot()
  })

  it('properly activates callback when clicked', () => {
    expect(onClick).not.toHaveBeenCalled()
    burger.simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
