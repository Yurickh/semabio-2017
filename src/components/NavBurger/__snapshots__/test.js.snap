// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBurger should render correctly 1`] = `
<div
  className="nav-burger "
  onClick={[Function]}
/>
`;
